package NServer;

option go_package = "a.yandex-team.ru/kernel/server/protos;kernel_server";

message THttpServerConfig {
    optional uint32 Port                  = 1  [default = 17000];
    optional uint32 Threads               = 2  [default = 30];
    optional uint32 MaxQueueSize          = 3  [default = 2000];
    optional uint32 MaxFQueueSize         = 4  [default = 10];
    optional bool   CompressionEnabled    = 5  [default = false];
    optional bool   KeepAliveEnabled      = 6  [default = true];
    optional uint32 RequestTTL            = 7  [default = 500]; // dynamic
    optional string LogFile               = 8;
    optional string ResponseTimeIntervals = 9  [default = "1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 150, 200, 250, 300, 350, 400, 450, 500, 1000, 5000"];
    optional string DynamicConfigPath     = 10 [default = "dynamicConfig"];
    optional uint32 ConfigCheckDelay      = 11 [default = 12]; // times in minute
    optional bool   EnableAppHost         = 12 [default = false];
    optional uint32 AppHostPort           = 13 [default = 17001];
    optional uint32 AppHostThreads        = 14 [default = 30];
    optional uint32 MaxConnections        = 15 [default = 400];
    repeated string BindAddress           = 16;
    optional bool   ElasticQueueEnabled   = 17 [default = false];
    optional string StatsPrefix           = 18 [default = ""];
    optional string ClientTimeout         = 19;
    optional bool   UseGrpcForApphost     = 20 [default = false];
    optional uint32 GrpcPort              = 21;
    optional uint32 GrpcThreadCount       = 22 [default = 4];
    optional uint32 ListenBacklog         = 23;
}
